scaleway/resource_compute_instance_volume.go

```diff
@@ -138,14 +168,43 @@ func resourceScalewayComputeInstanceVolumeDelete(d *schema.ResourceData, m inter
 		return err
 	}
 
-	err = instanceAPI.DeleteVolume(&instance.DeleteVolumeRequest{
-		VolumeID: id,
+	deleteRequest := &instance.DeleteVolumeRequest{
 		Zone:     zone,
-	})
-
-	if err != nil && !is404Error(err) {
-		return fmt.Errorf("couldn't delete volume: %v", err)
+		VolumeID: id,
 	}
 
-	return nil
+	err = resource.Retry(ServerRetryFuncTimeout, func() *resource.RetryError {
+		err := instanceAPI.DeleteVolume(deleteRequest)
+		if isSDKResponseError(err, http.StatusBadRequest, "a server is attached to this volume") {
```

Comment:
I don't like error checking by string or regex

Reply:
me too, do you see a better solution?